src/Filters/FiltersScope.php

```diff
@@ -3,17 +3,43 @@
 namespace Spatie\QueryBuilder\Filters;
 
 use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
 use Illuminate\Support\Str;
+use ReflectionObject;
 
 class FiltersScope implements Filter
 {
-    public function __invoke(Builder $query, $values, string $property)
+    public function __invoke(Builder $query, $values, string $property): Builder
     {
         $scope = Str::camel($property);
 
-        $values = Arr::wrap($values);
+        $values = array_values(Arr::wrap($values));
+        $values = $this->resolveParameters($query, $values, $scope);
 
-        $query->$scope(...array_values($values));
+        return $query->$scope(...$values);
+    }
+
+    protected function resolveParameters(Builder $query, $values, string $scope): array
+    {
+        $parameters = (new ReflectionObject($query->getModel()))
+            ->getMethod('scope' . ucfirst($scope))
+            ->getParameters();
+
+        foreach ($parameters as $parameter) {
+            if (! optional($parameter->getClass())->isSubclassOf(Model::class)) {
+                continue;
+            }
+
+            $model = $parameter->getClass()->newInstance();
+            $index = $parameter->getPosition() - 1;
+            $value = $values[$index];
+
+            if (is_numeric($value)) {
```

Comment:
Would it make sense for this line to take into account the `$keyType` property on the model instance?

Models using a string primary key (e.g UUID) can not be resolved currently.

Something like:
```php
$keyType = $model->getKeyType();

if ($keyType == 'string' || ($keyType == 'int' && is_numeric($value))
{
}
```

Reply:
@dominikb Yes, of course! Thank you for the tip :)